interface/wx/webrequest.h

```diff
@@ -0,0 +1,292 @@
+    HTTP and HTTPs up to HTTP/2.
+
+    <a href="https://docs.microsoft.com/en-us/windows/desktop/WinHttp/about-winhttp">
+    API Documentation</a>
+
+    @par NSURLSession on macOS and iOS
+
+    This API is available since macOS 10.9 and iOS 7.0.
+
+    <a href="https://developer.apple.com/documentation/foundation/urlsession">
+    API Documentation</a>
+
+    @par libcurl on Linux
+
+    This library is available on most Unix distributions.
+
+    <a href="https://curl.haxx.se/libcurl/">
+    API Documentation</a>
+
+    @beginEventEmissionTable{wxWebRequestEvent}
+    @event{wxEVT_WEBREQUEST_READY(id, func)}
+        The request data is ready to be read
+    @event{wxEVT_WEBREQUEST_FAILED(id, func)}
+        The request has failed
+    @event{wxEVT_WEBREQUEST_AUTH_REQUIRED(id, func)}
+        The request needs additional authentication to continue
+    @endEventTable
+
+    @since 3.1.2
+    @library{wxnet}
+    @category{net}
+
+    @see wxWebRequestSession
+*/
+class wxWebRequest: public wxEvtHandler
+{
+public:
+    /**
+        Create a new request instance
+
+        @param url The URL of the HTTP resource for this request
+        @param session If no wxWebRequestSession is give the default session will
+                       be used
+
+        @since 3.1.2
+    */
+    explicit wxWebRequest(const wxString& url, wxWebRequestSession* session = NULL);
+
+    /**
+
+        @since 3.1.2
+    */
+    ~wxWebRequest();
+
+    /**
+        Adds a request header send by this request.
+
+        @param name Name if the header
+        @param value String value of the header
+
+        @since 3.1.2
+    */
+    void AddRequestHeader(const wxString& name, const wxString& value);
+
+    /**
+        Set HTTP method.
+
+        Set <a href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html">common</a>
+        or expanded HTTP method.
+
+        The default method is GET unless request data is provided in which
+        case POST is the default.
+
+        @param method
+            HTTP method name, e.g. "GET".
+
+        @since 3.1.2
+    */
+    void SetRequestMethod(const wxString& method);
+
+    /**
+        Set the text to be posted to the server.
+
+        After a successful call to this method, the request will use HTTP @c
+        POST instead of the default @c GET when it's executed.
+
+        @param data
+            The data to post.
+        @param contentType
+            The value of HTTP "Content-Type" header, e.g. "text/html;
+            charset=UTF-8".
+
+        @since 3.1.2
+    */
+    void SetRequestData(const wxString& data, const wxString& contentType);
+
+    /**
+        Set the binary data to be posted to the server.
+
+        The next request will
+        be an HTTP @c POST instead of the default HTTP @c GET and the given @a
+        data will be posted as the body of this request.
+
+        @since 3.1.2
+    */
+    void SetRequestData(const wxInputStream*, const wxString& contentType);
```

Comment:
Again, I'm nervous about pointers and their ownership rules. Could you please document clearly for how long should this unnamed (why?) stream argument remain valid?

Also, does it really make sense for it to be null? If not, let's make it a reference.

Reply:
Made it to reference for now. It might have to become a `wxSharedPtr<>` when I've looked further into implementation. Unnamed was just a typo.